### DISCUSSION TITLE
text field component on submit action

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3900] [Ideas]
Hello,

Is there any plan to add submit action for text field component? Currently if user press enter or change focus, it will trigger submit and refresh the data provider, but in some case we may need more complex actions like update app state etc.

**Comment by shogunpurple** [comment]
HI @ggs331

I think this is a pretty nice idea - In general, what might be more flexible is to add a "Keyboard" event to certain components, where you could bind a certain key being hit, to some actions like updating app state as you mentioned.

Currently there are only `onClick` events in budibase for button actions, it could be good to expand the event system out more.
